lib/debug/repl.cpp

```diff
@@ -300,12 +322,68 @@ void Repl::addSpiCommands(const std::uint8_t bus)
       const core::Result result    = spi->write(register_address, data_ptr, 1);
       if (result == core::Result::kSuccess) {
         logger_.log(
-          core::LogLevel::kInfo, "Successful SPI write to device %d on %d", register_address, bus);
+          core::LogLevel::kDebug, "Successful SPI write to device %d on %d", register_address, bus);
       } else {
         logger_.log(core::LogLevel::kFatal, "Failed to write to SPI bus: %d", bus);
       }
     };
     addCommand(spi_write_byte_command);
   }
 }
-}  // namespace hyped::debug
+
+void Repl::addUartCommands(const std::uint8_t bus)
+{
+  const UartBus uart_bus   = static_cast<UartBus>(bus);
+  const auto optional_uart = io::Uart::create(logger_, uart_bus, BaudRate::kB38400);
+  if (!optional_uart) {
+    logger_.log(core::LogLevel::kFatal, "Failed to create UART instance on bus %d", bus);
+    return;
+  }
+  const auto uart = std::make_shared<io::Uart>(*optional_uart);
+  {
+    Command uart_read_command;
+    std::stringstream identifier;
+    identifier << "uart " << static_cast<int>(bus) << " read";
+    uart_read_command.name = identifier.str();
+    std::stringstream description;
+    description << "Read from UART bus " << static_cast<int>(bus);
+    uart_read_command.description = description.str();
+    uart_read_command.handler     = [this, uart, bus]() {
+      std::uint8_t read_buffer[255];
+      const core::Result result = uart->readBytes(read_buffer, 255);
+      if (result == core::Result::kSuccess) {
+        logger_.log(core::LogLevel::kDebug, "UART value from bus %d: %s", bus, read_buffer);
```

Comment:
do we want `read_buffer` to be parsed as a string here? surely the UART values are raw binary?

Reply:
we don't have a use case for this atm so I've kept it as bytes, will change if and when we do actually need UART

Reply:
i still disagree but i see your point